### PR TITLE
Check the return value of avifDecoderSetIOMemory.

### PR DESCRIPTION
**Pull request by vrabaud**
The API will soon be made no_discard.

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by asmorkalov** [comment]
> The API will soon be made no_discard.
What version do you mean?

**Review by asmorkalov — COMMENTED** [review]
:+1:

File: modules/imgcodecs/src/grfmt_avif.cpp [hunk 1]
```diff
@@ -170,6 +158,20 @@ bool AvifDecoder::checkSignature(const String &signature) const {
...
+  avifDecoder *decoder = avifDecoderCreate();
+  if (!decoder) return false;
+  OPENCV_AVIF_CHECK_STATUS(
+      avifDecoderSetIOMemory(
+          decoder, reinterpret_cast<const uint8_t *>(signature.c_str()),
+          signature.size()),
+      decoder);
+  decoder->io->sizeHint = 1e9;
+  const avifResult status = avifDecoderParse(decoder);
+  avifDecoderDestroy(decoder);
```
Comment (mshabunin) [inline modules/imgcodecs/src/grfmt_avif.cpp:171]: Won't `decoder` instance leak if we exit earlier due to failed `avifDecoderSetIOMemory` call?

Comment (vrabaud) [reply modules/imgcodecs/src/grfmt_avif.cpp:171]: You are totally right, I provided a fix in https://github.com/opencv/opencv/pull/24441. Thx !

**Pull request merged**